### PR TITLE
Feature: Pass arguments in `yarn algob run` cli 

### PR DESCRIPTION
**Pull request by rmeena840**
Feature: Pass arguments in `yarn algob run` cli.

New format: `yarn alob run --script script1.js arg1 arg2 arg3`

Screenshot 2022-11-21 at 1 28 34 PM

### CODE REVIEW

File: docs/guide/debugging-teal.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ If it displays the `tealdbg` help, you're good to go.
 
 Creating transaction data (via `goal --dryrun-dump` or SDK) could be a lengthy process, especially when using a transaction group. `Algob` provides an easy way to use debugger: by simply supplying the transactions as an input to the `TealDbg` method (same transaction parameters that we supply to [executeTx](https://algobuilder.dev/api/algob/modules.html#executeTransaction) to execute same transaction on network).
 
-NOTE: You use the `TealDbg` method in an algob script, which can be run using `algob deploy`/`algob run` commands.
+NOTE: You use the `TealDbg` method in an algob script, which can be run using `algob deploy`/`algob run --script  --script ` commands.
```
Comment (thdailong) [inline docs/guide/debugging-teal.md:29]: Why do we have double `--script` here?

File: docs/guide/deployer.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ Deployer class has the following modes:
 - Run Mode: In run mode user can access/read checkpoints, create logs but cannot write(create) checkpoints. Files placed in nested folders (non-direct children, eg: `scripts/transfer/run-script.js`) of `scripts/` folder are considered to be run in this mode.
   To run a script in the _run_ mode (the script will receive deployer instance in _run_ mode\_):
 
-        yarn run algob run scripts/transfer/run-script.js
+        yarn run algob run --script  --script  scripts/transfer/run-script.js
```
Comment (thdailong) [inline docs/guide/deployer.md:25]: Should be only one `--script` only according to the changes file above

File: .github/workflows/05-examples.yaml [hunk 3]
```diff
@@ -75,29 +75,29 @@ jobs:
         working-directory: ./examples/asa
         run: |
           pipenv run yarn algob deploy
-          pipenv run yarn algob run scripts/transfer/gold-contract-sc.js
-          pipenv run yarn algob run scripts/transfer/gold-delegated-lsig.js
-          pipenv run yarn algob run scripts/transfer/gold-to-john.js
-          pipenv run yarn algob run scripts/transfer/master-fund-john.js
-          pipenv run yarn algob run scripts/transfer/tesla-to-john.js
+          pipenv run yarn algob run --script  scripts/transfer/gold-contract-sc.js
```
Comment (thdailong) [inline .github/workflows/05-examples.yaml:78]:
```suggestion
          pipenv run yarn algob run --script scripts/transfer/gold-contract-sc.js
```
There should be one space only. The same for all `--script  `

File: packages/algob/src/internal/core/runtime-environment.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ export class Environment implements RuntimeEnv {
 	private readonly _extenders: EnvironmentExtender[];
 
 	/**
-	 * Initializes the Algob Runtime Environment and the given
+	 * Initializes the algob run --script time Environment and the given
```
Comment (thdailong) [inline packages/algob/src/internal/core/runtime-environment.ts:32]:
```suggestion
	 * Initializes the Algob Runtime Environment and the given
```

File: packages/algob/test/internal/util/scripts-runner.ts [hunk 5]
```diff
@@ -25,22 +25,22 @@ describe("Scripts runner", function () {
 	});
 
 	it("Should pass params to the script", async function () {
-		await runScript("./scripts/params-script.js", env, new DeployerRunMode(deployerCfg));
+		await runScript("./scripts/params-script.js", [], env, new DeployerRunMode(deployerCfg));
```
Comment (thdailong) [inline packages/algob/test/internal/util/scripts-runner.ts:28]: Could you test with an argument pass in the script?

File: packages/algob/src/builtin-tasks/run.ts [hunk 6]
```diff
@@ -56,6 +57,7 @@ function partitionIntoSorted(unsorted: string[]): string[][] {
 export async function runMultipleScripts(
```
Comment (thdailong) [inline packages/algob/src/builtin-tasks/run.ts:57]: I see what we are no longer running multiple scripts at a time. We should rename the function to like `runScripts`

**Review by thdailong — COMMENTED** [review]
There are much changes you need to make. 
+ Update changelog
+ Update sample project to show user how to use argument
+ If we can not run multiple scripts then you should update documents about it
+ Many `--script  ` code have two spaces which we should only need 1

File: docs/guide/debugging-teal.md (hunk 1 above)
Comment (rmeena840) [reply docs/guide/debugging-teal.md:29]: Format has changed now. Read description.

File: .github/workflows/05-examples.yaml (hunk 3 above)
Comment (rmeena840) [reply .github/workflows/05-examples.yaml]: Yes, done

File: packages/algob/src/internal/core/runtime-environment.ts (hunk 4 above)
Comment (rmeena840) [reply packages/algob/src/internal/core/runtime-environment.ts:32]: Why??

File: packages/algob/test/internal/util/scripts-runner.ts (hunk 5 above)
Comment (rmeena840) [reply packages/algob/test/internal/util/scripts-runner.ts:28]: Done

File: docs/guide/debugging-teal.md (hunk 1 above)
Comment (rmeena840) [reply docs/guide/debugging-teal.md]: Done

**Comment by rmeena840** [comment]
> There are much changes you need to make.
> 
> * Update changelog
> * Update sample project to show user how to use argument
> * If we can not run multiple scripts then you should update documents about it
> * Many `--script  ` code have two spaces which we should only need 1

1. Changelog updated
2. I have created task for this: https://www.pivotaltracker.com/story/show/183849580
3. I have created task for this: https://www.pivotaltracker.com/story/show/183849580
4. Done

File: packages/algob/src/builtin-tasks/run.ts (hunk 6 above)
Comment (rmeena840) [reply packages/algob/src/builtin-tasks/run.ts:57]: We need to think about it before we remove it. I have created a task for this.:https://www.pivotaltracker.com/story/show/183849598

File: docs/guide/user-script-execution.md [hunk 7]
```diff
@@ -8,15 +8,15 @@ layout: splash
 
 In algob, scripts are JS files stored in the `scripts` directory and are meant to interact with the blockchain (deploy ASA, ASC1, run transactions...).
 
-Scripts are run through `algob run` and `algob deploy` commands. `algob` is using `algob.config.js` to get information about network and accounts and is executing incremental scripts found in `scripts/`.
+Scripts are run through `algob run --script ` and `algob deploy` commands. `algob` is using `algob.config.js` to get information about network and accounts and is executing incremental scripts found in `scripts/`.
```
Comment (thdailong) [inline docs/guide/user-script-execution.md:11]:
```suggestion
Scripts are run through `algob run --script` and `algob deploy` commands. `algob` is using `algob.config.js` to get information about network and accounts and is executing incremental scripts found in `scripts/`.
```

File: docs/guide/deployer.md (hunk 2 above)
Comment (rmeena840) [reply docs/guide/deployer.md]: DOne

File: docs/guide/user-script-execution.md (hunk 7 above)
Comment (rmeena840) [reply docs/guide/user-script-execution.md]: Refresh, it's there.

File: .github/workflows/05-examples.yaml [hunk 8]
```diff
@@ -75,29 +75,29 @@ jobs:
         working-directory: ./examples/asa
         run: |
           pipenv run yarn algob deploy
-          pipenv run yarn algob run scripts/transfer/gold-contract-sc.js
-          pipenv run yarn algob run scripts/transfer/gold-delegated-lsig.js
-          pipenv run yarn algob run scripts/transfer/gold-to-john.js
-          pipenv run yarn algob run scripts/transfer/master-fund-john.js
-          pipenv run yarn algob run scripts/transfer/tesla-to-john.js
```
Comment (sczembor) [inline .github/workflows/05-examples.yaml:82]: Will it still work if we do not use `--scirpt`

**Pull request merged**